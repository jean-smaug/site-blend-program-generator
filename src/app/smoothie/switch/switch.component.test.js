import React from 'react';
import { shallow } from 'enzyme';

import SwitchComponent from './switch.component';

const conferences = [
  {
    timeEnd: 18,
    timeBegin: 16,
    date: 'day2',
    domain: 'biz',
    level: 'noob',
    description: 'Esse amet tempor aute sunt. Magna voluptate nulla duis ullamco amet nisi proident eiusmod duis.',
    name: 'nisi nostrud',
    id: '59724990bb9d86db92537dba',
  },
  {
    timeEnd: 12,
    timeBegin: 10,
    date: 'day2',
    domain: 'market',
    level: 'confirmed',
    description: 'Nisi elit aute amet magna. Mollit id ipsum amet cillum ea in duis duis officia irure dolor sint.',
    name: 'tempor anim',
    id: '5972499025b0804dc2b05651',
  },
  {
    timeEnd: 12,
    timeBegin: 10,
    date: 'day2',
    domain: 'tech',
    level: 'noob',
    description: 'Ex sit deserunt mollit dolore irure in ad nostrud minim enim. Consectetur aute aute culpa dolor irure velit consequat ullamco.',
    name: 'ad deserunt',
    id: '59724990716a4f824a36b3c5',
  },
];

describe('switch component', () => {
  it('should render the switch component', () => {
    const Switch = shallow(
      <SwitchComponent id={0} conferences={conferences} />,
    );
    expect(Switch).toMatchSnapshot();
    // expect(Switch.children.length).toBe(3);
  });

  // it('detect click button', () => {
  //   const Switch = shallow(
  //     <SwitchComponent id={0} conferences={conferences} />,
  //   );
  // });
});
