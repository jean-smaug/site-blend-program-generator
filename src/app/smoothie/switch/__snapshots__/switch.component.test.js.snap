// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`switch component should render the switch component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
        className="switcher"
    >
        <div>
            <a
                role="button"
            >
                nisi nostrud
            </a>
            <br />
        </div>
        <div>
            <a
                role="button"
            >
                tempor anim
            </a>
            <br />
        </div>
        <div>
            <a
                role="button"
            >
                ad deserunt
            </a>
            <br />
        </div>
    </div>
</div>,
  "nodes": Array [
    <div>
      <div
            className="switcher"
      >
            <div>
                  <a
                        role="button"
                  >
                        nisi nostrud
                  </a>
                  <br />
            </div>
            <div>
                  <a
                        role="button"
                  >
                        tempor anim
                  </a>
                  <br />
            </div>
            <div>
                  <a
                        role="button"
                  >
                        ad deserunt
                  </a>
                  <br />
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Switch
        conferences={
                Array [
                        Object {
                          "date": "day2",
                          "description": "Esse amet tempor aute sunt. Magna voluptate nulla duis ullamco amet nisi proident eiusmod duis.",
                          "domain": "biz",
                          "id": "59724990bb9d86db92537dba",
                          "level": "noob",
                          "name": "nisi nostrud",
                          "timeBegin": 16,
                          "timeEnd": 18,
                        },
                        Object {
                          "date": "day2",
                          "description": "Nisi elit aute amet magna. Mollit id ipsum amet cillum ea in duis duis officia irure dolor sint.",
                          "domain": "market",
                          "id": "5972499025b0804dc2b05651",
                          "level": "confirmed",
                          "name": "tempor anim",
                          "timeBegin": 10,
                          "timeEnd": 12,
                        },
                        Object {
                          "date": "day2",
                          "description": "Ex sit deserunt mollit dolore irure in ad nostrud minim enim. Consectetur aute aute culpa dolor irure velit consequat ullamco.",
                          "domain": "tech",
                          "id": "59724990716a4f824a36b3c5",
                          "level": "noob",
                          "name": "ad deserunt",
                          "timeBegin": 10,
                          "timeEnd": 12,
                        },
                      ]
        }
        id={0}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Switch {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "conferences": Array [
            Object {
              "date": "day2",
              "description": "Esse amet tempor aute sunt. Magna voluptate nulla duis ullamco amet nisi proident eiusmod duis.",
              "domain": "biz",
              "id": "59724990bb9d86db92537dba",
              "level": "noob",
              "name": "nisi nostrud",
              "timeBegin": 16,
              "timeEnd": 18,
            },
            Object {
              "date": "day2",
              "description": "Nisi elit aute amet magna. Mollit id ipsum amet cillum ea in duis duis officia irure dolor sint.",
              "domain": "market",
              "id": "5972499025b0804dc2b05651",
              "level": "confirmed",
              "name": "tempor anim",
              "timeBegin": 10,
              "timeEnd": 12,
            },
            Object {
              "date": "day2",
              "description": "Ex sit deserunt mollit dolore irure in ad nostrud minim enim. Consectetur aute aute culpa dolor irure velit consequat ullamco.",
              "domain": "tech",
              "id": "59724990716a4f824a36b3c5",
              "level": "noob",
              "name": "ad deserunt",
              "timeBegin": 10,
              "timeEnd": 12,
            },
          ],
          "id": 0,
        },
        "refs": Object {},
        "selectConference": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div
                    className="switcher"
          >
                    <div>
                              <a
                                        role="button"
                              >
                                        nisi nostrud
                              </a>
                              <br />
                    </div>
                    <div>
                              <a
                                        role="button"
                              >
                                        tempor anim
                              </a>
                              <br />
                    </div>
                    <div>
                              <a
                                        role="button"
                              >
                                        ad deserunt
                              </a>
                              <br />
                    </div>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <div
                    className="switcher"
          >
                    <div>
                              <a
                                        role="button"
                              >
                                        nisi nostrud
                              </a>
                              <br />
                    </div>
                    <div>
                              <a
                                        role="button"
                              >
                                        tempor anim
                              </a>
                              <br />
                    </div>
                    <div>
                              <a
                                        role="button"
                              >
                                        ad deserunt
                              </a>
                              <br />
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Switch
    conferences={
        Array [
            Object {
              "date": "day2",
              "description": "Esse amet tempor aute sunt. Magna voluptate nulla duis ullamco amet nisi proident eiusmod duis.",
              "domain": "biz",
              "id": "59724990bb9d86db92537dba",
              "level": "noob",
              "name": "nisi nostrud",
              "timeBegin": 16,
              "timeEnd": 18,
            },
            Object {
              "date": "day2",
              "description": "Nisi elit aute amet magna. Mollit id ipsum amet cillum ea in duis duis officia irure dolor sint.",
              "domain": "market",
              "id": "5972499025b0804dc2b05651",
              "level": "confirmed",
              "name": "tempor anim",
              "timeBegin": 10,
              "timeEnd": 12,
            },
            Object {
              "date": "day2",
              "description": "Ex sit deserunt mollit dolore irure in ad nostrud minim enim. Consectetur aute aute culpa dolor irure velit consequat ullamco.",
              "domain": "tech",
              "id": "59724990716a4f824a36b3c5",
              "level": "noob",
              "name": "ad deserunt",
              "timeBegin": 10,
              "timeEnd": 12,
            },
          ]
    }
    id={0}
/>,
}
`;
